projects/packages/my-jetpack/_inc/components/product-cards-section/index.tsx

```diff
@@ -90,17 +92,52 @@ const DisplayItems: FC< DisplayItemsProps > = ( { slugs } ) => {
 
 interface ProductCardsSectionProps {
 	noticeMessage: ReactNode;
+	siteIsRegistered?: boolean;
 }
 
-const ProductCardsSection: FC< ProductCardsSectionProps > = ( { noticeMessage } ) => {
-	const { ownedProducts = [], unownedProducts = [] } =
+const ProductCardsSection: FC< ProductCardsSectionProps > = ( {
+	noticeMessage,
+	siteIsRegistered,
+} ) => {
+	const { ownedProducts: inntialOwnedProducts = [], unownedProducts: inntialUnownedProducts = [] } =
 		getMyJetpackWindowInitialState( 'lifecycleStats' );
+	const [ siteOwnedProducts, setSiteOwnedProducts ] = useValueStore(
+		'siteOwnedProducts',
+		inntialOwnedProducts
+	);
+	const [ siteUnownedProducts, setSiteUnownedProducts ] = useValueStore(
+		'siteUnownedProducts',
+		inntialUnownedProducts
+	);
```

Comment:
Would it be a good idea to keep these values within `useProductsByOwnership()` hook? In a way, view component doesn't need to know if it's served initial or refetched data.

Reply:
That's a good point. However, if I put this into the `useProductsByOwnership` hook, it won't return results until the call finishes fetching data, making the initial data redundant. 

I could add the window data as `placeholderData` for the query. Do you think this would be better?

Reply:
Here is the draft of what I was thinking. Curious what you're thinking about it - let me know!

```javascript
const useProductsByOwnership = () => {
	const [ productsOwnership, setProductsOwnership ] = useValueStore( 'productsOwnership', {
		ownedProducts: getMyJetpackWindowInitialState( 'lifecycleStats' ).ownedProducts,
		unownedProducts: getMyJetpackWindowInitialState( 'lifecycleStats' ).unownedProducts,
	} );

	const { data, refetch, isLoading } = useFetchProductsByOwnership();

	useEffect( () => {
		if ( ! isLoading && data ) {
			const { ownedProducts = [], unownedProducts = [] } = data;
			setProductsOwnership( { ownedProducts, unownedProducts } );
		}
	}, [ data, isLoading, setProductsOwnership ] );

	return {
		productsOwnership,
		isLoading,
		refetchOwnershipData: refetch,
	};
};
```

In this case, we would be able to have two instances of `useProductsByOwnership()` - one in `projects/packages/my-jetpack/_inc/components/welcome-flow/ConnectionStep.tsx` and one as we have now - but if we refetch data in ConnectionStep - the other instance will get the updated data automatically.

Maybe that would be better than creating useEffect, that might run more times than we actually need.

Reply:
Thanks for the suggestion. I like it. The downside is that we need to add refetch to all places that can make a connection, but this won't be as common as showing the modules, so in the end we benefit by removing useEffect from the original place.

Reply:
Happy that it worked!

> The downside is that we need to add refetch to all places that can make a connection

Thanks for outlining that, indeed it's an additional logic, but I believe it might actually give us more control over when we refetch things.